### ISSUE TITLE
Validate log levels

### ISSUE THREAD

**Issue by rahulreddy** [scality/S3#268]
This issue is to address any inconsistent levels of logging from high to low level layers.
For example, we want to establish if we want to log any non 404s in https://github.com/scality/S3/blob/master/lib/metadata/wrapper.js
https://github.com/scality/S3/blob/master/lib/metadata/bucketclient/backend.js.

I am not convinced that we have to log with `error` level in low layers than `debug` or `trace`. Ideally we should only be logging at dump level only once per error path in a request.

**Comment by rahulreddy** [comment]
Moving the conversation from #253 for better tracking.

**Comment by DavidPineauScality** [comment]
Agreed that logging at dump-level should happen mostly once per code path. That being said, it's harming no-one if it happens multiple times, as the full logging history is flushed out when the dump happens (so no double-dump anyways, you'll just dump the newer log entries since the previous dump)

Anyways, we should document this properly for future reference (even only as comments in the wrapper or lower layers?)

Thanks for caring for this.

My personal stance would be the following:

In the lower layers:
 - When there is no doubt that the error will generate a 500 at S3-level, dump it right away
 - When there is an ambiguity on a specific error code (such as 404), debug on it

In the upper layers:
 - When the underlying code provides no doubt that the error will generate a 500 in S3, do nothing (or for safety dump again?)
 - When the possible error codes, are ambiguous, dump only if it will yield a 500, otherwise, debug.

I see that we have a different stance on it, but I guess that the most important is to have a clear agreed stance, and make the code consistent with it for the future.
